jupyter/jupyter.go

```diff
@@ -299,6 +296,36 @@ func (s *server) execute(c *conn, req *message) error {
 		return s.publishIO("error", content, req)
 	}
 
+	r, w, err := os.Pipe()
+	if err != nil {
+		return errResp(err)
+	}
+
+	session.Stdout = w
+	session.Stderr = w
+	stdout := os.Stdout
+	os.Stdout = w
+	stderr := os.Stderr
+	os.Stderr = w
+
+	done := make(chan struct{})
+	buf := new(bytes.Buffer)
+	go func() {
+		io.Copy(buf, r)
+		close(done)
+	}()
+
+	err = session.Exec([]byte(reqContent["code"].(string)))
+
+	w.Close()
+	<-done
+	r.Close()
```

Comment:
Okay, I don't think i fully understood what was going on.  I'm getting closer (I think).  Thanks for the patience.